test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTagHelpersTests.cs

```diff
@@ -0,0 +1,103 @@
+
+            // The K runtime compiles every file under compiler/resources as a resource at runtime with the same name
+            // as the file name, in order to update a baseline you just need to change the file in that folder.
+            var expectedContent =
+                    await _resourcesAssembly.ReadResourceAsStringAsync
+                                     ("compiler/resources/MvcTagHelpersWebSite.MvcTagHelperTest." + action + ".html");
+
+            // Act
+            // The host is not important as everything runs in memory and tests are isolated from each other.
+            var response = await client.GetAsync("http://localhost/MvcTagHelperTest/" + action);
+            var responseContent = await response.Content.ReadAsStringAsync();
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(expectedMediaType, response.Content.Headers.ContentType);
+
+            if (antiForgeryPath != null)
+            {
+                var forgeryToken = AntiForgeryTestHelper.RetrieveAntiForgeryToken(responseContent, antiForgeryPath);
+                expectedContent = string.Format(expectedContent, forgeryToken);
+            }
+            Assert.Equal(expectedContent.Trim(), responseContent.Trim());
+        }
+
+        [Fact]
+        public async Task ValidationTagHelpers_GeneratesExpectedSpansAndDivs()
+        {
+            // Arrange
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+            var response = await client.GetAsync("http://localhost/MvcTagHelperTest/Customer");
+            var responseContent = await response.Content.ReadAsStringAsync();
+
+            var expectedContent =
+                    await _resourcesAssembly.ReadResourceAsStringAsync
+                                     ("compiler/resources/MvcTagHelpersWebSite.Customer.Index.html");
+
+            var forgeryCookie = AntiForgeryTestHelper.RetrieveAntiForgeryCookie(response);
+            var forgeryToken = AntiForgeryTestHelper.RetrieveAntiForgeryToken(responseContent, "/Customer/Customer");
+
+            var request = new HttpRequestMessage(HttpMethod.Post, "http://localhost/Customer/Customer");
+            request.Headers.Add("Cookie", "__RequestVerificationToken=" + forgeryCookie);
+            var nameValueCollection = new List<KeyValuePair<string, string>>
+            {
+                new KeyValuePair<string,string>("__RequestVerificationToken", forgeryToken),
+                new KeyValuePair<string,string>("Number", string.Empty),
+                new KeyValuePair<string,string>("Name", string.Empty),
+                new KeyValuePair<string,string>("Email", string.Empty),
+                new KeyValuePair<string,string>("PhoneNumber", string.Empty),
+                new KeyValuePair<string,string>("Password", string.Empty)
+            };
```

Comment:
I'll remove the others, but I think I should keep _requestverifiationtoken since they return the different results depending on the existence of that token.